gateway/src/apicast/policy/apicast/apicast.lua

```diff
@@ -98,6 +102,11 @@ function _M:access(context)
 end
 
 function _M:content(context)
+  if not context[self].upstream then
+    ngx.log(ngx.WARN, "Upstream server not found for this request")
```

Comment:
I'm not sure here, but I think that something like this need to be in place, but I'm not sure. 

```
  ngx.status = 504
  ngx.print('')
  return exit()
```

Reply:
Why is this `if` needed? Shouldn't we treat this case like we did up until now with the assert below?

Reply:
The thing is if you have 3 services and one is not correctly set, APICast will die and other services will be affected. 

With this solution, the current service will not work but others will be not affected at all.

Reply:
I think that we need a test case showing that apicast no longer crashes in that case.